File: nomad-samples/python/basic-live-channel-1.1/live_input/get_live_input.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

'''
 * GET Live Input
 *
 * Note: Live Inputs are direct calls,
 *       they are not indexed, thus, the Search API cannot be used.
 *
 * @param {string} AUTH_TOKEN    | Authentication token
 * @param {string} INPUT_ID      | The ID of the live input to get
 *
 * @returns JSON Object containing the requested live input
'''
async def get_live_input(AUTH_TOKEN, INPUT_ID):
    # Check for valid parameters
    if (not AUTH_TOKEN or not INPUT_ID):
        raise Exception("Get Live Input: Invalid API call")


    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        # Send the request
        RESPONSE = requests.get(ADMIN_URL + "/liveInput/" + INPUT_ID, headers= HEADERS)

        

        # If not found return None
        if (not RESPONSE.ok):
            raise Exception("Live Input with ID " + INPUT_ID + " was not found")
        
        

        return json.loads(RESPONSE.text)


    except:
        await api_exception_handler(RESPONSE, f"Get Live Input with ID {INPUT_ID} failed")

File: nomad-samples/admin/js/advanced-live-channel-1.1/live-input/get-live-input.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * GET Live Input
 *
 * Note: Live Inputs are direct calls,
 *       they are not indexed, thus, the Search API cannot be used.
 *
 * @param {string} authToken    | Authorization token
 * @param {string} inputId      | The ID of the live input to get
 *
 * @returns JSON Object containing the requested live input
 */
export default async function getLiveInput(authToken, inputId) {
    // Check for valid parameters
    if (!authToken || !inputId) {
        throw new Error("Get Live Input: Invalid API call");
    }

    // Create header for the request
    const headers = new Headers();
    headers.append("Content-Type", "application/json");
    headers.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const response = await fetch(`${prjConstants.SERVER_URL}/liveInput/${inputId}`, {
        method: "GET",
        headers: headers
    });

    // Check for valid response
    if (response) {
        // Check for success
        if (response.ok) {
            // Return the JSON response
            return await response.json();
        }

        // If not found return null
        if (response.status === 404) {
            console.error(`Live Input with ID ${inputId} was not found`);
            return null;
        }
    }

    await apiExceptionHandler(response, `Get Live Input with ID ${inputId} failed`);
}
